# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsIContentInlines.h
# Commit: 5661dec1e128
# Full Hash: 5661dec1e128f48fc62be6ebc2774fc247b71fc3
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - Allow using flat tree order for point comparing in selection r=jjaschke,smaug,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231588
# ==============================================================================

diff -r 15ccf86bce5a -r 5661dec1e128 dom/base/nsIContentInlines.h
--- a/dom/base/nsIContentInlines.h	Wed May 07 15:05:08 2025 +0000
+++ b/dom/base/nsIContentInlines.h	Wed May 07 15:05:09 2025 +0000
@@ -144,9 +144,8 @@
   return ::GetFlattenedTreeParentNode<nsINode::eForStyle>(this);
 }
 
-inline nsIContent* nsINode::GetFlattenedTreeParentNodeForSelection() const {
-  nsINode* parent = ::GetFlattenedTreeParentNode<nsINode::eForSelection>(this);
-  return (parent && parent->IsContent()) ? parent->AsContent() : nullptr;
+inline nsINode* nsINode::GetFlattenedTreeParentNodeForSelection() const {
+  return ::GetFlattenedTreeParentNode<nsINode::eForSelection>(this);
 }
 
 inline bool nsINode::NodeOrAncestorHasDirAuto() const {